### ISSUE TITLE
Build failure: pkgs.pkgsCross.riscv64.gcc14

### ISSUE THREAD

**Issue by YooLc** [NixOS/nixpkgs#351905]
### Steps To Reproduce

1. build cross compile gcc 14 of RISC-V 64 on a x86-64 machine, you can reproduce it using the following `flake.nix`

```nix
{
  inputs = {
    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
  };

  outputs = { self, nixpkgs }:
    let
      pkgs = import nixpkgs {
        system = "x86_64-linux";
      };
    in
    {
      devShell.x86_64-linux = pkgs.pkgsCross.riscv64.mkShell {
        nativeBuildInputs = with pkgs; [
          pkgs.pkgsCross.riscv64.gcc14
        ];
      };
    };
}
```

### Build log

```
error: builder for '/nix/store/x9wnygf7vkdb9hjjb9d71yizgxh02krs-gcc-riscv64-unknown-linux-gnu-14.2.0.drv' failed with exit code 2;
       last 25 log lines:
       > libtool: compile:  riscv64-unknown-linux-gnu-g++ -O2 -I/nix/store/2srlsvl5aqvhk3lnqizc2fm3j0fdzp3k-glibc-riscv64-unknown-linux-gnu-2.40-36-dev/include -B/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib/ -idirafter /nix/store/2srlsvl5aqvhk3lnqizc2fm3j0fdzp3k-glibc-riscv64-unknown-linux-gnu-2.40-36-dev/include -idirafter /nix/store/sq78a6hq5bms0wp2crvwysypqr2kvr68-riscv64-unknown-linux-gnu-gcc-13.3.0/lib/gcc/riscv64-unknown-linux-gnu/13.3.0/include-fixed -Wl,-rpath,/nix/store/q2wm7gnwvhls3yj156xw0dh1x974gmn1-gcc-riscv64-unknown-linux-gnu-14.2.0-lib/lib -Wl,-L/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib -Wl,-rpath -Wl,/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib -Wl,-dynamic-linker=/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib/ld-linux-riscv64-lp64d.so.1 --sysroot=/ -I/build/gcc-14.2.0/libstdc++-v3/../libgcc -I/build/build/riscv64-unknown-linux-gnu/libstdc++-v3/include/riscv64-unknown-linux-gnu -I/build/build/riscv64-unknown-linux-gnu/libstdc++-v3/include -I/build/gcc-14.2.0/libstdc++-v3/libsupc++ -std=gnu++20 -D_GLIBCXX_SHARED -fno-implicit-templates -Wall -Wextra -Wwrite-strings -Wcast-qual -Wabi=2 -fdiagnostics-show-location=once -ffunction-sections -fdata-sections -frandom-seed=tzdb.lo -fimplicit-templates -O2 -I/nix/store/2srlsvl5aqvhk3lnqizc2fm3j0fdzp3k-glibc-riscv64-unknown-linux-gnu-2.40-36-dev/include -B/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib/ -idirafter /nix/store/2srlsvl5aqvhk3lnqizc2fm3j0fdzp3k-glibc-riscv64-unknown-linux-gnu-2.40-36-dev/include -idirafter /nix/store/sq78a6hq5bms0wp2crvwysypqr2kvr68-riscv64-unknown-linux-gnu-gcc-13.3.0/lib/gcc/riscv64-unknown-linux-gnu/13.3.0/include-fixed -Wl,-rpath,/nix/store/q2wm7gnwvhls3yj156xw0dh1x974gmn1-gcc-riscv64-unknown-linux-gnu-14.2.0-lib/lib -Wl,-L/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib -Wl,-rpath -Wl,/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib -Wl,-dynamic-linker=/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib/ld-linux-riscv64-lp64d.so.1 -I. -c ../../../../../gcc-14.2.0/libstdc++-v3/src/c++20/tzdb.cc  -fPIC -DPIC -D_GLIBCXX_SHARED -o tzdb.o
       > /nix/store/717iy55ncqs0wmhdkwc5fg2vci5wbmq8-bash-5.2p32/bin/bash ../../libtool --tag CXX --tag disable-shared   --mode=link riscv64-unknown-linux-gnu-g++ -O2 -I/nix/store/2srlsvl5aqvhk3lnqizc2fm3j0fdzp3k-glibc-riscv64-unknown-linux-gnu-2.40-36-dev/include -B/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib/ -idirafter /nix/store/2srlsvl5aqvhk3lnqizc2fm3j0fdzp3k-glibc-riscv64-unknown-linux-gnu-2.40-36-dev/include -idirafter /nix/store/sq78a6hq5bms0wp2crvwysypqr2kvr68-riscv64-unknown-linux-gnu-gcc-13.3.0/lib/gcc/riscv64-unknown-linux-gnu/13.3.0/include-fixed -Wl,-rpath,/nix/store/q2wm7gnwvhls3yj156xw0dh1x974gmn1-gcc-riscv64-unknown-linux-gnu-14.2.0-lib/lib -Wl,-L/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib -Wl,-rpath -Wl,/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib -Wl,-dynamic-linker=/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib/ld-linux-riscv64-lp64d.so.1 --sysroot=/    -Wl,-O1 -Wl,-z,relro -Wl,--gc-sections  -std=gnu++20 -prefer-pic -D_GLIBCXX_SHARED -fno-implicit-templates  -Wall -Wextra -Wwrite-strings -Wcast-qual -Wabi=2  -fdiagnostics-show-location=once   -ffunction-sections -fdata-sections  -frandom-seed=libc++20convenience.la  -fimplicit-templates  '-Wl,-rpath,/nix/store/q2wm7gnwvhls3yj156xw0dh1x974gmn1-gcc-riscv64-unknown-linux-gnu-14.2.0-lib/lib' '-Wl,-L/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib' '-Wl,-rpath' '-Wl,/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib' '-Wl,-dynamic-linker=/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib/ld-linux-riscv64-lp64d.so.1' -o libc++20convenience.la  tzdb.lo sstream-inst.lo
       > libtool: link: riscv64-unknown-linux-gnu-ar --plugin /nix/store/sq78a6hq5bms0wp2crvwysypqr2kvr68-riscv64-unknown-linux-gnu-gcc-13.3.0/libexec/gcc/riscv64-unknown-linux-gnu/13.3.0/liblto_plugin.so rc .libs/libc++20convenience.a  tzdb.o sstream-inst.o
       > libtool: link: riscv64-unknown-linux-gnu-ranlib --plugin /nix/store/sq78a6hq5bms0wp2crvwysypqr2kvr68-riscv64-unknown-linux-gnu-gcc-13.3.0/libexec/gcc/riscv64-unknown-linux-gnu/13.3.0/liblto_plugin.so .libs/libc++20convenience.a
       > libtool: link: ( cd ".libs" && rm -f "libc++20convenience.la" && ln -s "../libc++20convenience.la" "libc++20convenience.la" )
       > make[5]: Leaving directory '/build/build/riscv64-unknown-linux-gnu/libstdc++-v3/src/c++20'
       > Making all in c++23
       > make[5]: Entering directory '/build/build/riscv64-unknown-linux-gnu/libstdc++-v3/src/c++23'
       > /nix/store/717iy55ncqs0wmhdkwc5fg2vci5wbmq8-bash-5.2p32/bin/bash ../../libtool --tag CXX --tag disable-shared   --mode=compile riscv64-unknown-linux-gnu-g++ -O2 -I/nix/store/2srlsvl5aqvhk3lnqizc2fm3j0fdzp3k-glibc-riscv64-unknown-linux-gnu-2.40-36-dev/include -B/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib/ -idirafter /nix/store/2srlsvl5aqvhk3lnqizc2fm3j0fdzp3k-glibc-riscv64-unknown-linux-gnu-2.40-36-dev/include -idirafter /nix/store/sq78a6hq5bms0wp2crvwysypqr2kvr68-riscv64-unknown-linux-gnu-gcc-13.3.0/lib/gcc/riscv64-unknown-linux-gnu/13.3.0/include-fixed -Wl,-rpath,/nix/store/q2wm7gnwvhls3yj156xw0dh1x974gmn1-gcc-riscv64-unknown-linux-gnu-14.2.0-lib/lib -Wl,-L/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib -Wl,-rpath -Wl,/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib -Wl,-dynamic-linker=/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib/ld-linux-riscv64-lp64d.so.1 --sysroot=/   -I/build/gcc-14.2.0/libstdc++-v3/../libgcc -I/build/build/riscv64-unknown-linux-gnu/libstdc++-v3/include/riscv64-unknown-linux-gnu -I/build/build/riscv64-unknown-linux-gnu/libstdc++-v3/include -I/build/gcc-14.2.0/libstdc++-v3/libsupc++   -std=gnu++23 -prefer-pic -D_GLIBCXX_SHARED -fno-implicit-templates  -Wall -Wextra -Wwrite-strings -Wcast-qual -Wabi=2  -fdiagnostics-show-location=once   -ffunction-sections -fdata-sections  -frandom-seed=stacktrace.lo  -fimplicit-templates -O2 -I/nix/store/2srlsvl5aqvhk3lnqizc2fm3j0fdzp3k-glibc-riscv64-unknown-linux-gnu-2.40-36-dev/include -B/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib/ -idirafter /nix/store/2srlsvl5aqvhk3lnqizc2fm3j0fdzp3k-glibc-riscv64-unknown-linux-gnu-2.40-36-dev/include -idirafter /nix/store/sq78a6hq5bms0wp2crvwysypqr2kvr68-riscv64-unknown-linux-gnu-gcc-13.3.0/lib/gcc/riscv64-unknown-linux-gnu/13.3.0/include-fixed -Wl,-rpath,/nix/store/q2wm7gnwvhls3yj156xw0dh1x974gmn1-gcc-riscv64-unknown-linux-gnu-14.2.0-lib/lib -Wl,-L/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib -Wl,-rpath -Wl,/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib -Wl,-dynamic-linker=/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib/ld-linux-riscv64-lp64d.so.1  -c -o stacktrace.lo ../../../../../gcc-14.2.0/libstdc++-v3/src/c++23/stacktrace.cc
       > /nix/store/717iy55ncqs0wmhdkwc5fg2vci5wbmq8-bash-5.2p32/bin/bash ../../libtool --tag CXX --tag disable-shared   --mode=compile riscv64-unknown-linux-gnu-g++ -O2 -I/nix/store/2srlsvl5aqvhk3lnqizc2fm3j0fdzp3k-glibc-riscv64-unknown-linux-gnu-2.40-36-dev/include -B/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib/ -idirafter /nix/store/2srlsvl5aqvhk3lnqizc2fm3j0fdzp3k-glibc-riscv64-unknown-linux-gnu-2.40-36-dev/include -idirafter /nix/store/sq78a6hq5bms0wp2crvwysypqr2kvr68-riscv64-unknown-linux-gnu-gcc-13.3.0/lib/gcc/riscv64-unknown-linux-gnu/13.3.0/include-fixed -Wl,-rpath,/nix/store/q2wm7gnwvhls3yj156xw0dh1x974gmn1-gcc-riscv64-unknown-linux-gnu-14.2.0-lib/lib -Wl,-L/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib -Wl,-rpath -Wl,/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib -Wl,-dynamic-linker=/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib/ld-linux-riscv64-lp64d.so.1 --sysroot=/   -I/build/gcc-14.2.0/libstdc++-v3/../libgcc -I/build/build/riscv64-unknown-linux-gnu/libstdc++-v3/include/riscv64-unknown-linux-gnu -I/build/build/riscv64-unknown-linux-gnu/libstdc++-v3/include -I/build/gcc-14.2.0/libstdc++-v3/libsupc++   -std=gnu++23 -prefer-pic -D_GLIBCXX_SHARED -fno-implicit-templates  -Wall -Wextra -Wwrite-strings -Wcast-qual -Wabi=2  -fdiagnostics-show-location=once   -ffunction-sections -fdata-sections  -frandom-seed=print.lo  -fimplicit-templates -O2 -I/nix/store/2srlsvl5aqvhk3lnqizc2fm3j0fdzp3k-glibc-riscv64-unknown-linux-gnu-2.40-36-dev/include -B/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib/ -idirafter /nix/store/2srlsvl5aqvhk3lnqizc2fm3j0fdzp3k-glibc-riscv64-unknown-linux-gnu-2.40-36-dev/include -idirafter /nix/store/sq78a6hq5bms0wp2crvwysypqr2kvr68-riscv64-unknown-linux-gnu-gcc-13.3.0/lib/gcc/riscv64-unknown-linux-gnu/13.3.0/include-fixed -Wl,-rpath,/nix/store/q2wm7gnwvhls3yj156xw0dh1x974gmn1-gcc-riscv64-unknown-linux-gnu-14.2.0-lib/lib -Wl,-L/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib -Wl,-rpath -Wl,/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib -Wl,-dynamic-linker=/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib/ld-linux-riscv64-lp64d.so.1  -std=gnu++26 -c ../../../../../gcc-14.2.0/libstdc++-v3/src/c++23/print.cc
       > libtool: compile:  riscv64-unknown-linux-gnu-g++ -O2 -I/nix/store/2srlsvl5aqvhk3lnqizc2fm3j0fdzp3k-glibc-riscv64-unknown-linux-gnu-2.40-36-dev/include -B/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib/ -idirafter /nix/store/2srlsvl5aqvhk3lnqizc2fm3j0fdzp3k-glibc-riscv64-unknown-linux-gnu-2.40-36-dev/include -idirafter /nix/store/sq78a6hq5bms0wp2crvwysypqr2kvr68-riscv64-unknown-linux-gnu-gcc-13.3.0/lib/gcc/riscv64-unknown-linux-gnu/13.3.0/include-fixed -Wl,-rpath,/nix/store/q2wm7gnwvhls3yj156xw0dh1x974gmn1-gcc-riscv64-unknown-linux-gnu-14.2.0-lib/lib -Wl,-L/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib -Wl,-rpath -Wl,/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib -Wl,-dynamic-linker=/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib/ld-linux-riscv64-lp64d.so.1 --sysroot=/ -I/build/gcc-14.2.0/libstdc++-v3/../libgcc -I/build/build/riscv64-unknown-linux-gnu/libstdc++-v3/include/riscv64-unknown-linux-gnu -I/build/build/riscv64-unknown-linux-gnu/libstdc++-v3/include -I/build/gcc-14.2.0/libstdc++-v3/libsupc++ -std=gnu++23 -D_GLIBCXX_SHARED -fno-implicit-templates -Wall -Wextra -Wwrite-strings -Wcast-qual -Wabi=2 -fdiagnostics-show-location=once -ffunction-sections -fdata-sections -frandom-seed=stacktrace.lo -fimplicit-templates -O2 -I/nix/store/2srlsvl5aqvhk3lnqizc2fm3j0fdzp3k-glibc-riscv64-unknown-linux-gnu-2.40-36-dev/include -B/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib/ -idirafter /nix/store/2srlsvl5aqvhk3lnqizc2fm3j0fdzp3k-glibc-riscv64-unknown-linux-gnu-2.40-36-dev/include -idirafter /nix/store/sq78a6hq5bms0wp2crvwysypqr2kvr68-riscv64-unknown-linux-gnu-gcc-13.3.0/lib/gcc/riscv64-unknown-linux-gnu/13.3.0/include-fixed -Wl,-rpath,/nix/store/q2wm7gnwvhls3yj156xw0dh1x974gmn1-gcc-riscv64-unknown-linux-gnu-14.2.0-lib/lib -Wl,-L/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib -Wl,-rpath -Wl,/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib -Wl,-dynamic-linker=/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib/ld-linux-riscv64-lp64d.so.1 -c ../../../../../gcc-14.2.0/libstdc++-v3/src/c++23/stacktrace.cc  -fPIC -DPIC -D_GLIBCXX_SHARED -o stacktrace.o
       > libtool: compile:  riscv64-unknown-linux-gnu-g++ -O2 -I/nix/store/2srlsvl5aqvhk3lnqizc2fm3j0fdzp3k-glibc-riscv64-unknown-linux-gnu-2.40-36-dev/include -B/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib/ -idirafter /nix/store/2srlsvl5aqvhk3lnqizc2fm3j0fdzp3k-glibc-riscv64-unknown-linux-gnu-2.40-36-dev/include -idirafter /nix/store/sq78a6hq5bms0wp2crvwysypqr2kvr68-riscv64-unknown-linux-gnu-gcc-13.3.0/lib/gcc/riscv64-unknown-linux-gnu/13.3.0/include-fixed -Wl,-rpath,/nix/store/q2wm7gnwvhls3yj156xw0dh1x974gmn1-gcc-riscv64-unknown-linux-gnu-14.2.0-lib/lib -Wl,-L/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib -Wl,-rpath -Wl,/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib -Wl,-dynamic-linker=/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib/ld-linux-riscv64-lp64d.so.1 --sysroot=/ -I/build/gcc-14.2.0/libstdc++-v3/../libgcc -I/build/build/riscv64-unknown-linux-gnu/libstdc++-v3/include/riscv64-unknown-linux-gnu -I/build/build/riscv64-unknown-linux-gnu/libstdc++-v3/include -I/build/gcc-14.2.0/libstdc++-v3/libsupc++ -std=gnu++23 -D_GLIBCXX_SHARED -fno-implicit-templates -Wall -Wextra -Wwrite-strings -Wcast-qual -Wabi=2 -fdiagnostics-show-location=once -ffunction-sections -fdata-sections -frandom-seed=print.lo -fimplicit-templates -O2 -I/nix/store/2srlsvl5aqvhk3lnqizc2fm3j0fdzp3k-glibc-riscv64-unknown-linux-gnu-2.40-36-dev/include -B/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib/ -idirafter /nix/store/2srlsvl5aqvhk3lnqizc2fm3j0fdzp3k-glibc-riscv64-unknown-linux-gnu-2.40-36-dev/include -idirafter /nix/store/sq78a6hq5bms0wp2crvwysypqr2kvr68-riscv64-unknown-linux-gnu-gcc-13.3.0/lib/gcc/riscv64-unknown-linux-gnu/13.3.0/include-fixed -Wl,-rpath,/nix/store/q2wm7gnwvhls3yj156xw0dh1x974gmn1-gcc-riscv64-unknown-linux-gnu-14.2.0-lib/lib -Wl,-L/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib -Wl,-rpath -Wl,/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib -Wl,-dynamic-linker=/nix/store/44z53k46jmnnldr8zshw5lc3pcikwdwj-glibc-riscv64-unknown-linux-gnu-2.40-36/lib/ld-linux-riscv64-lp64d.so.1 -std=gnu++26 -c ../../../../../gcc-14.2.0/libstdc++-v3/src/c++23/print.cc  -fPIC -DPIC -D_GLIBCXX_SHARED -o print.o
       > riscv64-unknown-linux-gnu-g++: error: unrecognized command-line option '-std=gnu++26'; did you mean '-std=gnu++20'?
       > make[5]: *** [Makefile:747: print.lo] Error 1
       > make[5]: *** Waiting for unfinished jobs....
       > make[5]: Leaving directory '/build/build/riscv64-unknown-linux-gnu/libstdc++-v3/src/c++23'
       > make[4]: *** [Makefile:797: all-recursive] Error 1
       > make[4]: Leaving directory '/build/build/riscv64-unknown-linux-gnu/libstdc++-v3/src'
       > make[3]: *** [Makefile:577: all-recursive] Error 1
       > make[3]: Leaving directory '/build/build/riscv64-unknown-linux-gnu/libstdc++-v3'
       > make[2]: *** [Makefile:502: all] Error 2
       > make[2]: Leaving directory '/build/build/riscv64-unknown-linux-gnu/libstdc++-v3'
       > make[1]: *** [Makefile:11897: all-target-libstdc++-v3] Error 2
       > make[1]: Leaving directory '/build/build'
       > make: *** [Makefile:1061: all] Error 2
       For full logs, run 'nix log /nix/store/x9wnygf7vkdb9hjjb9d71yizgxh02krs-gcc-riscv64-unknown-linux-gnu-14.2.0.drv'.
error: 1 dependencies of derivation '/nix/store/kjb5h661xlqi4n177w2kmdgps3whg92b-gcc-wrapper-14.2.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/pn5k7sh8fc7l76lzphf5vc8icaj0l1vx-nix-shell-riscv64-unknown-linux-gnu-env.drv' failed to build
```

### Additional context

The error in the log above is `riscv64-unknown-linux-gnu-g++: error: unrecognized command-line option '-std=gnu++26'; did you mean '-std=gnu++20'?`, maybe it would be useful to check out what lead to this build failure.

### Metadata

```bash
 - system: `"x86_64-linux"`
 - host os: `Linux 6.11.5, NixOS, 24.11 (Vicuna), 24.11.20241025.18536bf`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.9`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/wb6agba4kfsxpbnb5hzlq58vkjzvbsk6-source
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by zimward** [comment]
it seems to be known as there is a closed pr targeting the same issue #303514 .

**Comment by trofi** [comment]
Yeah, ideally cross-building gcc should use the same version of gcc cross-compiler to be able to generate code that target libraries expect. Historically `nixpkgs` was not very diligent about that.

WDYT of the following fix:
- https://github.com/NixOS/nixpkgs/pull/352821
